src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -106,22 +109,18 @@ impl<Src: OpendalSource> OpendalReader<Src> {
                 let parser =
                     ByteStreamSourceParserImpl::create(self.parser_config.clone(), source_ctx)
                         .await?;
-                chunk_stream = if need_nd_streaming(&self.parser_config.specific.encoding_config) {
-                    Box::pin(parser.into_stream(nd_streaming::split_stream(data_stream)))
-                } else {
-                    Box::pin(parser.into_stream(data_stream))
```

Comment:
Actually this `else` branch should never be reached, otherwise `parser::into_chunk_stream_inner` cannot correctly parse the messages.